pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/IngestionDelayTracker.java

```diff
@@ -174,6 +188,22 @@ private long getIngestionDelayMs(long ingestionTimeMs) {
     return agedIngestionDelayMs;
   }
 
+  private long getPartitionOffsetLag(IngestionOffsets offset) {
```

Comment:
This comment applies to time based reporting as well and can be taken as follow up.  

Should this belong to individual consumer types instead of common code just like how getCurrentPartitionLagState is interfaced out, because reporting the metric for all types here might not make any sense ? 

Like you said, only for Kafka, this is relevant.

Reply:
Yeah that's a good suggestion. I also don't feel that casting the LongOffset is a good idea tbh.